src/vse_sync_pp/analyzers/ts2phc.py

```diff
@@ -0,0 +1,58 @@
+### SPDX-License-Identifier: GPL-2.0-or-later
+
+"""Analyze ts2phc log messages"""
+
+from .analyzer import Analyzer
+
+class TimeErrorAnalyzer(Analyzer):
+    """Analyze time error"""
+    id_ = 'ts2phc/time-error'
+    parser = id_
+    def __init__(self, config):
+        super().__init__(config)
+        # required system time output accuracy
+        accuracy = config.requirement('time-error-in-locked-mode/ns')
+        # limit on inaccuracy at observation point
+        limit = config.parameter('time-error-limit/%')
+        # exclusive upper bound on absolute time error for any sample
+        self._unacceptable = accuracy * limit / 100
+        # samples in the initial transient period are ignored
+        self._transient = config.parameter('transient-period/s')
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+    def prepare(self, rows):
```

Comment:
when we arrive to `prepare`  we should check we have enough samples in the log file to generate our data in canonical format. 
If I enter a file input with not enough samples I'm getting IndexError:out of range in return (rows[0]._fields, rows)

Reply:
I just noticed this too. Thinking...

Reply:
```>>> df = DataFrame.from_records((), columns=())
>>> df
Empty DataFrame
Columns: []
Index: []```

Reply:
I used `(False, "no data")` for (reason, result) to indicate this. Maybe the start of a convention?